socs/agents/hwp_encoder/agent.py

```diff
@@ -511,6 +512,36 @@ def __init__(self, agent_obj, port=8080):
                                  agg_params=agg_params)
         self.parser = EncoderParser(beaglebone_port=self.port)
 
+    def restart(self, session, params):
+        """restart()
+
+        **Task** - Restarts the beaglebone process
+
+        Notes:
+            The most recent data collected is stored in the session data in the
+            structure:
+
+                >>> response.session['response']
+                {'result': True,
+                 'log': ["Restart command response: Success"]}
+        """
+        if self.ip == 'None':
+            return False, "Could not restart process because beaglebone ip is not defined"
+
+        _restart_socket = socket.socket(socket.AF_INET, socket.SOCK_STREAM)
+        _restart_socket.connect((self.ip, 5656))
+        _restart_socket.sendall(('reset\n').encode())
+        time.sleep(0.5)
+
+        resp = _restart_socket.recv(4096).decode().strip()
+        log = f'Restart command response: {resp}'
+        result = True if resp == "Success" else False
+        _restart_socket.close()
+        time.sleep(10)
```

Comment:
Is this sleep just to give the beaglebone time to restart?

Reply:
Yeah. I'm not sure how long the restart process on the beaglebone takes (It's likely only a couple seconds) so I just put a conservative sleep time just to be sure that the agent doesn't try to restart when a restart is already running